### PR TITLE
Copter: Add GCS failsafe arming check, tones, and lights

### PR DESCRIPTION
**Pull request by Pedals2Paddles**
This adds the GCS failsafe to the AP_Notify library tones and lights.  For the most part, it's doing the same lights as the radio failsafe. There are only so many blinks and colors, so I just has use the same thing as the radio. Open to better ideas on this.

This also adds the GCS failsafe to the AP_Arming library pre-arm checks. You will get pre-arm warnings if the failsafe is active, and it will not arm.

The GCS failsafe is disabled by default in master now, and presumably will be defaulted to disabled in Copter 4.0.1 as well.  So this shouldn't cause a major undesired annoyance now.

_As a reminder, if no GCS has ever been connected, the GCS failsafe remains inactive._

**At this time, it appears copter is the only vehicle that will utilize this**.  Other vehicle types are not setting the new notify flag yet.  The notify flag didn't exist until recently anyway. So if other vehicle types wish to do this too, they would need to start setting the new AP_Notify flag for GCS failsafe.

### CODE REVIEW

**Comment by Pedals2Paddles** [comment]
Rebased.  Now that copter's GCS failsafe will be disabled by default, this should be a non-annoying change. And for those who make the positive decision to enable the failsafe, this will make it complete.

**Comment by tridge** [comment]
looks good to me, but I think Randy should review

File: libraries/AP_Arming/AP_Arming.h [hunk 1]
```diff
@@ -34,6 +34,7 @@ class AP_Arming {
         ARMING_CHECK_MISSION     = (1U << 14),
         ARMING_CHECK_RANGEFINDER = (1U << 15),
         ARMING_CHECK_CAMERA      = (1U << 16),
+        ARMING_CHECK_GCS         = (1U << 17),
```
Comment (tridge) [inline libraries/AP_Arming/AP_Arming.h]: this seems unused?

File: ArduCopter/AP_State.cpp [hunk 2]
```diff
@@ -74,6 +74,9 @@ void Copter::set_failsafe_radio(bool b)
 void Copter::set_failsafe_gcs(bool b)
 {
     failsafe.gcs = b;
+    
+    // update AP_Notify
+        AP_Notify::flags.failsafe_gcs = b;
```
Comment (WickedShell) [inline ArduCopter/AP_State.cpp]: Nitpick: Inconsistent whitespace.

Comment (rmackay9) [inline ArduCopter/AP_State.cpp]: small white space issue here..

**Comment by rmackay9** [comment]
As a follow-up we should add this to other vehicles as well but that's out of scope for this PR

**Comment by Pedals2Paddles** [comment]
Thanks for the reviews. I'll get it buttoned up this week.

**Comment by Pedals2Paddles** [comment]
Fixed everything. Of course I had left out the part that actually performs the arming check, so that's in now.  It's checking the same way it checks radio failsafe, which is looking at the AP_Notify flag.  

I don't like using the notify library as a databus for other functions, but there isn't any other means to present that information globally.  But we're doing that in numerous places right now absent a better way, which I think is a subject for another PR on another day.

**Comment by rmackay9** [comment]
@Pedals2Paddles, sorry to keep asking for changes but I think we should put this into the Copter specific AP_Arming class.  I'm also not keen on having the notify library acting as a databus so until the failsafes are moved into the vehicle class, let's keep them vehicle specific.

I was hoping we could get away without adding another bit to the ARMING_CHECKS parameter .. I guess we could add it to "System" although it's not a good fit with Terrain, Scripting and internal errors which are the other items in that category.  If we really do want to have a new bit then let's call it "Telem".

**Comment by WickedShell** [comment]
@rmackay9 I'd support not having a bit to turn it off actually. It's in the category of error that if you are working around it you are immediately at risk of failsaing/losing the protection. Just my opinion though, I'm fine with it being included in something. We do only have 7 more bits left though before we will have to make a CHECK2 parameter which would just be a bit ugly (although maybe inevitable?).

**Comment by Pedals2Paddles** [comment]
> I'd support not having a bit to turn it off actually. It's in the category of error that if you are working around it you are immediately at risk of failing/losing the protection.

I actually agree with this.  Disabling the failsafe would be the way to get it to arm. At least then you've taken steps to know you're disabling a safety system. I find when people disable arming checks, it's taken less seriously because they think it's a false positive.

But the same logic would dictate dumping the battery and RC failsafe arming checks which already exist today.  Someone else can take on that can of worms!

**Comment by Pedals2Paddles** [comment]
> sorry to keep asking for changes but I think the check should be put into the Copter specific AP_Arming class. I'm also not keen on having the notify library acting as a databus so until the failsafes are moved into the vehicle class, let's keep them vehicle specific.

@rmackay9 I will move the check into copter arming, and use the copter failsafe bool.  Much cleaner for now.

@rmackay9 @WickedShell it seems like the use of a bit in `ARMING_CHECKS` is precedent today. The other failsafes have arming check bits as well, so consistency would dictate putting in a bit for the GCS failsafe.  But functionally I think it's pointless and nobody would notice it missing either. I agree with MDB, in that the other failsafes also shouldn't have a way to disable the arming checks (RC, battery, etc).   The correct way to get around it is to disable the failsafe, not disable the arming check of the the failsafe state.  But that's a topic for another PR on another day.  I'm happy to leave the bit for GCS out of this one though.

**Comment by Pedals2Paddles** [comment]
Moved out of the AP_Arming library and into Copter arming. No `arming_check` bit due to it being rather pointless. Tested in SITL.

**Comment by Pedals2Paddles** [comment]
Let's do one final review/chat on the 2020-01-20 dev call.

File: libraries/AP_Notify/OreoLED_I2C.cpp [hunk 3]
```diff
@@ -159,6 +163,20 @@ bool OreoLED_I2C::mode_failsafe_radio()
 }
 
 
+// Procedure for when Pixhawk is in GCS failsafe
```
Comment (tridge) [inline libraries/AP_Notify/OreoLED_I2C.cpp]: minor nitpick, not all the world is Pixhawk

Comment (Pedals2Paddles) [reply libraries/AP_Notify/OreoLED_I2C.cpp]: Removed all of those references from the file.

**Comment by rmackay9** [comment]
Merged, thanks!

**Comment by rmackay9** [comment]
This is included in Copter-4.0.2-rc1

**Pull request merged**